pagefaultgames/pokerogue
# Add ESLint Rule to Enforce No Trailing Spaces

## What are the changes?
I have added a new rule in the ESLint configuration file (`.eslintrc`). This rule enforces the removal of trailing spaces at the end of lines, even on blank lines and lines containing comments.

## Why am I doing these changes?
The introduction of this rule is to maintain code quality and consistency across the codebase. Trailing spaces can cause unnecessary diffs and may lead to confusion during code reviews. By enforcing this rule, we ensure that the code remains clean and readable.

## What did change?
Before the change, trailing spaces at the end of lines, including blank lines and lines with comments, were allowed. After the change, any trailing spaces will trigger an ESLint error, prompting the developer to remove them. 

The specific rule added to the `.eslintrc` file is:
```json
"no-trailing-spaces": ["error", {
  "skipBlankLines": false,
  "ignoreComments": false
}]
```
### Screenshots/Videos
Not applicable, as this change does not impact the visual user experience.

## How to test the changes?
To test the changes, follow these steps:
1. Check out the branch with the new ESLint rule.
2. Run ESLint on the codebase using the command `npm run eslint`.
3. Ensure that ESLint raises an error for any lines with trailing spaces, including blank lines and lines containing comments.
4. Correct any errors by removing trailing spaces and re-run ESLint to verify that the errors are resolved.

## Checklist
